src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,22 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Check whether there are nullable nulls in at least one part of the foreign key.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            if ($schema->isNullable($field) && $entity->get($field) === null) {
+                return true;
+            }
```

Comment:
This might be contrived, but could we still run into issues with 3 column primary keys? For example, 2 of the columns are not nullable, and the third is. Presently this rule would pass and the insertion/update would fail. Should we check that the null count is equal to or less than the number of nullable columns?